### PR TITLE
Fix several autofocus issues in the back end

### PR DESCRIPTION
**Pull request by leofeyer**
The PR fixes two autofocus issues:

* `input.autocomplete` is an empty string if not set and never `undefined`, therefore the first matching field was never focused.
* Our access keys did not work in TinyMCE (see #5067).

### CODE REVIEW

File: core-bundle/contao/templates/backend/be_tinyMCE.html5 [hunk 1]
```diff
@@ -16,26 +16,29 @@ window.tinymce && tinymce.init({
...
-  setup: function(editor) {
+  setup: editor => {
     editor.getElement().removeAttribute('required');
+    document.querySelectorAll('[accesskey]').forEach(el => {
+      editor.addShortcut('access+' + el.accessKey, el.id, () => el.click());
+    });
   },
-  init_instance_callback: function(editor) {
+  init_instance_callback: editor => {
+    console.log(document.activeElement.id);
```
Comment (aschempp) [inline core-bundle/contao/templates/backend/be_tinyMCE.html5]: that's probably a leftover?

File: core-bundle/contao/templates/backend/be_tinyMCE.html5 [hunk 2]
```diff
@@ -16,26 +16,29 @@ window.tinymce && tinymce.init({
...
-    file_picker_callback: function (callback, value, meta) {
+    file_picker_callback: (callback, value, meta) => {
       Backend.openModalSelector({
         'id': 'tl_listing',
         'title': document.getElement('.tox-dialog__title').get('text'),
         'url': Contao.routes.backend_picker + '?context=' + (meta.filetype == 'file' ? 'link' : 'file') + '&amp;extras[fieldType]=radio&amp;extras[filesOnly]=true&amp;extras[source]=<?= $this->source ?>&amp;value=' + value + '&amp;popup=1',
-        'callback': function(table, val) {
-          callback(val.join(','));
-        }
+        'callback': (table, val) => callback(val.join(','))
```
Comment (aschempp) [inline core-bundle/contao/templates/backend/be_tinyMCE.html5]: using array functions will drop support for IE completely, is that intentional?

File: core-bundle/contao/templates/backend/be_tinyMCE.html5 (hunk 1 above)
Comment (leofeyer) [reply core-bundle/contao/templates/backend/be_tinyMCE.html5]: Yes. Removed in 9819d6cfbba5771b096a646f6199792d92a7d0a8.

File: core-bundle/contao/templates/backend/be_tinyMCE.html5 (hunk 2 above)
Comment (leofeyer) [reply core-bundle/contao/templates/backend/be_tinyMCE.html5]: <del>No, not really. Reverted in 9819d6cfbba5771b096a646f6199792d92a7d0a8.</del>

We are already using arrow functions, so IE is no longer supported anyway:

https://github.com/contao/contao/blob/4af387e032da7259112bbc5ea8f235a9f3b26286/core-bundle/src/Resources/contao/drivers/DC_Table.php#L2095-L2098

File: core-bundle/contao/templates/backend/be_tinyMCE.html5 [hunk 3]
```diff
@@ -16,26 +16,28 @@ window.tinymce && tinymce.init({
...
+    document.querySelectorAll('[accesskey]').forEach(function(el) {
+      editor.addShortcut('access+' + el.accessKey, el.id, () => el.click());
+    });
   },
   init_instance_callback: function(editor) {
     if (document.activeElement && document.activeElement.id && document.activeElement.id == editor.id) {
       editor.editorManager.get(editor.id).focus();
     }
-    editor.on('focus', function() { Backend.getScrollOffset(); });
+    editor.on('focus', () => Backend.getScrollOffset());
```
Comment (aschempp) [inline core-bundle/contao/templates/backend/be_tinyMCE.html5:31]: this is still an arrow function

Comment (m-vo) [reply core-bundle/contao/templates/backend/be_tinyMCE.html5:31]: That's what the commit message says, no?
image

**Review by ausi — APPROVED** [review]

File: core-bundle/contao/templates/backend/be_tinyMCE.html5 (hunk 3 above)
Comment (aschempp) [reply core-bundle/contao/templates/backend/be_tinyMCE.html5:31]: but its not the case for all other functions 🤷

Comment (leofeyer) [reply core-bundle/contao/templates/backend/be_tinyMCE.html5:31]: See https://github.com/contao/contao/pull/5266#discussion_r968344179

**Pull request merged**